using/src/implementations.rs

```diff
@@ -0,0 +1,208 @@
+
+        assert_eq!(
+            impls.find(&identity, &Version(1, 0)).collect::<Vec<_>>(),
+            vec![
+                (&Version(1, 0), &"1.0"),
+                (&Version(1, 2), &"1.2"),
+                (&Version(1, 3), &"1.3"),
+                (&Version(1, 5), &"1.5"),
+            ]
+        );
+
+        assert_eq!(
+            impls.find(&identity, &Version(1, 0)).bounds(),
+            Some(((&Version(1, 0), &"1.0"), (&Version(1, 5), &"1.5"),))
+        );
+
+        assert_eq!(
+            impls.find(&identity, &Version(2, 1)).collect::<Vec<_>>(),
+            vec![(&Version(2, 99), &"2.99"),]
+        );
+    }
+
+    #[test]
+    fn it_ignores_unrelated_specs() {
+        let identity = "https://spec.example.com/specA";
+        let unrelated = "https://spec.example.com/B";
+        let impls = Implementations::new()
+            .provide(identity, Version(0, 9), "too small")
+            .provide(identity, Version(2, 99), "2.99")
+            .provide(unrelated, Version(1, 3), "unrelated 1.3")
+            .provide(identity, Version(1, 0), "1.0")
+            .provide(unrelated, Version(1, 2), "unrelated 1.2")
+            .provide(identity, Version(1, 2), "1.2")
+            .provide(unrelated, Version(1, 5), "unrelated 1.5")
+            .provide(identity, Version(1, 3), "1.3")
+            .provide(identity, Version(1, 5), "1.5")
+            .provide(unrelated, Version(2, 0), "2.0")
+            .provide(identity, Version(2, 0), "2.0");
+        assert_eq!(
+            impls.find(&identity, &Version(1, 0)).collect::<Vec<_>>(),
+            vec![
+                (&Version(1, 0), &"1.0"),
+                (&Version(1, 2), &"1.2"),
+                (&Version(1, 3), &"1.3"),
+                (&Version(1, 5), &"1.5"),
+            ]
+        );
+        assert_eq!(
+            impls.find(&identity, &Version(2, 1)).next(),
+            Some((&Version(2, 99), &"2.99"))
+        );
+    }
+
+    #[test]
+    fn it_treats_each_zerodot_version_as_mutually_incompatible() {
+        let identity = "https://spec.example.com/specA";
+        let impls = Implementations::new()
+            .provide(identity, Version(0, 0), "0.0")
+            .provide(identity, Version(0, 1), "0.1")
+            .provide(identity, Version(0, 2), "0.0")
+            .provide(identity, Version(0, 3), "0.1")
+            .provide(identity, Version(0, 99), "0.99");
+        assert_eq!(
+            impls.find(&identity, &Version(0, 1)).last(),
+            Some((&Version(0, 1), &"0.1"))
+        );
+        assert_eq!(
+            impls.find(&identity, &Version(0, 99)).next(),
+            Some((&Version(0, 99), &"0.99"))
+        );
+        assert_eq!(
+            impls.find(&identity, &Version(0, 1)).next(),
+            Some((&Version(0, 1), &"0.1"))
+        );
+        assert_eq!(
+            impls.find(&identity, &Version(0, 99)).last(),
+            Some((&Version(0, 99), &"0.99"))
+        );
```

Comment:
Nit: the ordering of these assertions doesn't appear to follow a pattern and seems a bit awkward to follow. Additionally, it might be more clear if we `assert_eq` against the full `.collect()` rather than just first and last which are assumed to be the only two items in the iterator.